### PR TITLE
Add tools page and update dropdown and sidebars

### PR DESCRIPTION
**Pull request by laurapacilio**
Add a documentation page that lists the HashiCorp tools that work with Terraform. This effort will serve the following goals:

- Improve SEO for the terraform.io site.
- Promote Terraform projects that can solve user pain points.
- Show how HashiCorp products can work together to support modern infrastructure practices.

### CODE REVIEW

File: content/docs/terraform-tools.mdx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+
+## Manage Infrastructure
+
+|Name | Maintainer |Description| Tutorial |
+|-----|------------|-----------|----------|
+|[Kubernetes Operator for Terraform](https://github.com/hashicorp/terraform-k8s)| HashiCorp | Manage cloud and on-prem infrastructure through a Kubernetes custom resource definition (CRD) and Terraform Cloud. | [Deploy Infrastructure with the Terraform Cloud Operator for Kubernetes](https://learn.hashicorp.com/tutorials/terraform/kubernetes-operator) |
+| [Terraform ServiceNow Catalog](/docs/cloud/integrations/service-now/index.html) | HashiCorp |  Create Terraform configuration to order Service Items, create workspaces, and perform Terraform runs from ServiceNow.| |
+| [Consul Terraform Sync (Sync)](https://github.com/hashicorp/consul-terraform-sync) | HashiCorp | Automatically request network infrastructure updates to accommodate Consul service registrations. Works with A10 ADC, F5 BIG IP, Palo Alto Networks, and more. | [Network Infrastructure Automation (NIA)](https://learn.hashicorp.com/collections/consul/network-infrastructure-automation)  |
```
Comment (judithpatudith) [inline content/docs/terraform-tools.mdx:27]:
```suggestion
| [Consul Terraform Sync (CTS)](https://github.com/hashicorp/consul-terraform-sync) | HashiCorp | Automatically request network infrastructure updates to accommodate Consul service registrations. Works with A10 ADC, F5 BIG IP, Palo Alto Networks, and more. | [Network Infrastructure Automation (NIA)](https://learn.hashicorp.com/collections/consul/network-infrastructure-automation)  |
```

File: content/docs/terraform-tools.mdx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+
+## Automate Terraform
+
+|Name | Maintainer |Description| Tutorial |
+|-----|------------|-----------|----------|
+| [`terraform-exec`](https://github.com/hashicorp/terraform-exec) | HashiCorp | Use the Go module to construct and run CLI commands. Refer to the `terraform-json` repository (below) for documentation about the return values. |  |
+| [`terraform-json`](https://github.com/hashicorp/terraform-json) | HashiCorp | Use the data types and helper functions in this repository to work with JSON-formatted Terraform output.|  |
+| [Terraform Orb](https://circleci.com/developer/orbs/orb/circleci/terraform) | Third Party | Deploy your Terraform infrastructure via a CircleCI pipeline. CircleCI Orbs let you package, ship, and re-use configurations. | [Deploy Terraform infrastructure with CircleCI](https://learn.hashicorp.com/tutorials/terraform/circle-ci) |
```
Comment (judithpatudith) [inline content/docs/terraform-tools.mdx:36]:
```suggestion
| [CircleCI Orb](https://circleci.com/developer/orbs/orb/circleci/terraform) | Third Party | Deploy your Terraform infrastructure via a CircleCI pipeline. CircleCI Orbs let you package, ship, and re-use configurations. | [Deploy Terraform infrastructure with CircleCI](https://learn.hashicorp.com/tutorials/terraform/circle-ci) |
```

File: content/docs/terraform-tools.mdx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+|[GitHub Actions for Terraform](https://github.com/hashicorp/setup-terraform) | Third Party | Automate Terraform in your GitHub Actions workflow. After you have used the action, subsequent steps in the same job can run Terraform commands using the GitHub Actions run syntax. | [Automate Terraform with GitHub Actions](https://learn.hashicorp.com/tutorials/terraform/github-actions) |
+
+## Security and Compliance
+
+|Name | Maintainer |Description| Tutorial |
+|-----|------------|-----------|----------|
+| [Terraform Cloud for Splunk](/docs/cloud/integrations/splunk/index.html) | HashiCorp | Pull audit logs from Terraform Cloud into Splunk to identify the most active policies, understand significant changes in resource operations, and filter actions by specific users within your organization. | |
```
Comment (judithpatudith) [inline content/docs/terraform-tools.mdx:43]:
```suggestion
| [Splunk and Terraform Cloud](/docs/cloud/integrations/splunk/index.html) | HashiCorp | Pull audit logs from Terraform Cloud into Splunk to identify the most active policies, understand significant changes in resource operations, and filter actions by specific users within your organization. | |
```

**Review by judithpatudith — APPROVED** [review]
This is awesome! Could we remove "Terraform" anywhere it's the first words of the first column so that the tool names are more scannable? 

And also, could we add a section for the TFE provider to the Automate Terraform table? 
- Name: TFE provider
- Description: 
- Manage Terraform Cloud or Terraform Enterprise with Terraform itself. Works with organizations, workspaces, teams, variables, run triggers, policy sets, and more.
- Link: https://registry.terraform.io/providers/hashicorp/tfe/latest

Other than that it looks awesome! 👍🏻

File: content/docs/terraform-tools.mdx [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+
+## HashiCorp Products
+You can use HashiCorp products together to enable self-service automation across teams and organizations.
+
+|Name | Maintainer | Description| Tutorial |
+|-----|------------|-----------|----------|
+| [Vault provider](https://registry.terraform.io/providers/hashicorp/vault/latest/docs) | HashiCorp | Vault lets you secure, store, and tightly control access to tokens, passwords, certificates, encryption keys, and other sensitive data. Use this provider to read from, write to, and configure Vault.  | [Inject Secrets into Terraform Using the Vault Provider](https://learn.hashicorp.com/tutorials/cloud/terraform-hcp-provider-vault)|
+|[Consul Provider](https://registry.terraform.io/providers/hashicorp/consul/latest) | HashiCorp | [Consul](https://www.consul.io/) provides service-based networking for dynamic infrastructure. Use this provider to add external services to Consul, work with the key/value store, and more, using Terraform.  | [Use Terraform to Register External Services](https://learn.hashicorp.com/tutorials/consul/terraform-consul-provider)|
```
Comment (bflad) [inline content/docs/terraform-tools.mdx:65]: Should we also plug the Boundary, HCP, HCS, and TFC/E providers?

https://registry.terraform.io/providers/hashicorp/boundary/latest
https://registry.terraform.io/providers/hashicorp/hcp/latest
https://registry.terraform.io/providers/hashicorp/hcs/latest
https://registry.terraform.io/providers/hashicorp/tfe/latest

File: content/docs/terraform-tools.mdx [hunk 5]
```diff
@@ -0,0 +1,70 @@
+---
+page_title: Terraform Tools
+description: >-
+  Use these tools with Terraform to solve a broad range of infrastructure and DevOps challenges.
+
+---
```
Comment (im2nguyen) [inline content/docs/terraform-tools.mdx:6]: remove extra space

```suggestion
  Use these tools with Terraform to solve a broad range of infrastructure and DevOps challenges.
---
```

File: content/docs/terraform-tools.mdx [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+# Terraform Tools
+Terraform is part of a rich infrastructure and DevOps ecosystem. The tools below extend Terraform’s functionality or pair with Terraform to solve a broad range of infrastructure challenges.
+
+## Develop and Validate Configuration
+
+|Name | Maintainer |Description| Tutorial |
+|-----|------------|-----------|----------|
```
Comment (im2nguyen) [inline content/docs/terraform-tools.mdx:14]: Is there a way for us to make the columns more consistent (by combining columns or something like that)? The columns aren't evenly spaced and imo makes the table more difficult to skim ([Preview link](https://terraform-website-git-add-terraform-tools-hashicorp.vercel.app/docs/terraform-tools))

<details>
  <summary>screenshot showing misaligned tables</summary>

image

</details>

Comment (im2nguyen) [reply content/docs/terraform-tools.mdx:14]: for example, something like this is clean and easier to skim

image

File: content/docs/terraform-tools.mdx (hunk 4 above)
Comment (im2nguyen) [inline content/docs/terraform-tools.mdx:65]: currently only consul is linked

if we link out to consul, we should also link out to the other products to be more consistent 

image

**Review by im2nguyen — APPROVED** [review]
Overall looks great! Pre-approving so you're not blocked, but strongly agree with both Judith and Brian -- we should include the HCP provider and the other products (Boundary, HCS, HCP and TFE)

For the HCP provider, i would make a minor adjustment to Judith's description

> Manage Terraform Cloud or Terraform Enterprise with Terraform itself. Create, modify and delete organizations, workspaces, teams, variables, run triggers, policy sets, and more.

---

This is a nit, but the tables are not consistently aligned which makes it slightly harder to skim. I think it might be cleaner if we convert it back into bullets. I trust your decision

image

File: content/docs/terraform-tools.mdx [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+|[Cloud Development Kit for Terraform (CDKTF)](/docs/cdktf/index.html)| HashiCorp | Use familiar programming languages to define and provision infrastructure. CDKTF currently supports TypeScript, Python, Java, C#, and Go (experimental).| [Write CDK for Terraform configurations](https://learn.hashicorp.com/collections/terraform/cdktf) |
+|[Terraform Visual Studio Code Extension](https://marketplace.visualstudio.com/items?itemName=HashiCorp.terraform) | HashiCorp | Enable syntax highlighting and other editing features to VS Code using the Terraform Language Server.| |
```
Comment (robin-norwood) [inline content/docs/terraform-tools.mdx:16]:
```suggestion
|[Terraform Visual Studio Code Extension](https://marketplace.visualstudio.com/items?itemName=HashiCorp.terraform) | HashiCorp | Enable syntax highlighting and other editing features in VS Code using the Terraform Language Server.| |
```

File: content/docs/terraform-tools.mdx [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+| [Terraform ServiceNow Catalog](/docs/cloud/integrations/service-now/index.html) | HashiCorp |  Create Terraform configuration to order Service Items, create workspaces, and perform Terraform runs from ServiceNow.| |
+| [Consul Terraform Sync (Sync)](https://github.com/hashicorp/consul-terraform-sync) | HashiCorp | Automatically request network infrastructure updates to accommodate Consul service registrations. Works with A10 ADC, F5 BIG IP, Palo Alto Networks, and more. | [Network Infrastructure Automation (NIA)](https://learn.hashicorp.com/collections/consul/network-infrastructure-automation)  |
+
+
+## Automate Terraform
+
+|Name | Maintainer |Description| Tutorial |
+|-----|------------|-----------|----------|
+| [`terraform-exec`](https://github.com/hashicorp/terraform-exec) | HashiCorp | Use the Go module to construct and run CLI commands. Refer to the `terraform-json` repository (below) for documentation about the return values. |  |
```
Comment (robin-norwood) [inline content/docs/terraform-tools.mdx:34]:
```suggestion
| [`terraform-exec`](https://github.com/hashicorp/terraform-exec) | HashiCorp | Use this Go module to construct and run CLI commands. Refer to the `terraform-json` repository (below) for documentation about the return values. |  |
```

File: content/docs/terraform-tools.mdx [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+| [Sentinel](https://www.terraform.io/docs/cloud/sentinel/index.html) | HashiCorp |  Write and enforce fine-grained, logic-based policies for your infrastructure. Sentinel is an embedded policy-as-code framework integrated with the HashiCorp Enterprise products. | [Enforce Policy with Sentinel](https://learn.hashicorp.com/collections/terraform/policy) |
+
+
+## Develop Terraform Providers
+
+|Name | Maintainer |Description| Tutorial |
+|-----|------------|-----------|----------|
+| [Plugin SDK](/docs/extend/sdkv2-intro.html) | HashiCorp | Develop your own Terraform providers. | [Setup and Implement Read](https://learn.hashicorp.com/tutorials/terraform/provider-setup?in=terraform/providers) |
+|[Plugin Framework](/docs/plugin/framework/index.html) | HashiCorp | Develop your own Terraform providers. The framework provides improvements and additional features beyond what is available in the Plugin SDK. | [Implement Create and Read with the Terraform Plugin Framework](https://learn.hashicorp.com/tutorials/terraform/plugin-framework-create?in=terraform/providers) |
```
Comment (robin-norwood) [inline content/docs/terraform-tools.mdx:52]: This doesn't give the user much guidance as to where to go to first if they want to create a new provider.

**Review by robin-norwood — APPROVED** [review]
lgtm! A couple of suggestions. Also, it may be out of scope, but would it make sense to include Terraform cloud and terraform enterprise in this list of tools? From the description, it seems like they would count!

File: content/docs/terraform-tools.mdx (hunk 4 above)
Comment (laurapacilio) [reply content/docs/terraform-tools.mdx:65]: yes!! thank you so much - let me add these in :)

Comment (laurapacilio) [reply content/docs/terraform-tools.mdx:65]: Yes good call - I have added them!!

Comment (laurapacilio) [reply content/docs/terraform-tools.mdx:66]: YES thank you for catching this - i pushed up a change to fix :)

File: content/docs/terraform-tools.mdx (hunk 9 above)
Comment (laurapacilio) [reply content/docs/terraform-tools.mdx:52]: that's fair - this is a complex thing actually that I struggled with BUT I'm going to push up a tiny change to indicate that this ia "A new SDK to develop your own Terraform providers." This way it's clear from the first sentence that this is the most recent option, differentiating it a bit from the one above.

File: content/docs/terraform-tools.mdx (hunk 7 above)
Comment (laurapacilio) [reply content/docs/terraform-tools.mdx]: fixed locally!

File: content/docs/terraform-tools.mdx (hunk 6 above)
Comment (laurapacilio) [reply content/docs/terraform-tools.mdx:13]: I hear you - that does also bother me a bit BUT I still think that the tables make things easier to parse within each section. AND the names line up all together on the left side, so even though I don't LOVE that the table columns are not equal width, I think there is still value in the tables so I'm going to leave them. Thank you for calling this out though - I may talk to web dev about this in the future - this won't be the first time we want to use multiple tables on a page :)

**Pull request merged**